textile-utils-api-v1/src/main/java/net/textilemc/textile/utils/util/Identifier.java

```diff
@@ -0,0 +1,34 @@
+package net.textilemc.textile.utils.util;
+
+import net.minecraft.nbt.CompoundTag;
+
+public class Identifier {
+	private final String namespace;
+	private final String path;
+
+
+	public Identifier (String namespace, String path) {
+		this.namespace = namespace;
+		this.path = path;
+	}
+
+	public String getNamespace() {
+		return namespace;
+	}
+
+	public String getPath() {
+		return path;
+	}
+
+	public CompoundTag toTag(CompoundTag tag) {
+		CompoundTag subtag = new CompoundTag();
+		subtag.putString("namespace", this.getNamespace());
+		subtag.putString("path", this.getPath());
+		tag.put("nameId", subtag);
+		return tag;
+	}
+
+	public static Identifier fromTag(CompoundTag tag) {
+		return new Identifier(tag.getString("namespace"), tag.getString("path"));
+	}
```

Comment:
According to the `toTag` method just above, you save this to a `nameId` entry. You should then retrieve the identifier the same way.
```suggestion
	public static Identifier fromTag(CompoundTag tag) {
		String[] value = tag.getString("nameId").split(":");
		return new Identifier(value[0], value[1]);
	}
```

Reply:
serialization should not be done directly to a compound tag. nameId sounds weird.

Reply:
in fact, the compound tags in fromTag and toTag are meant to be of that object, so the only data you read is the data you’ve serialized

Reply:
and since Identifiers don’t use compound tags anymore, this should just be a toString and  a tryParse

Reply:
(so that the caller can choose any key name and so as to not violate the method contract)